libs/langgraph/src/tests/pregel.test-d.ts

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { expectTypeOf, it } from "vitest";
+import type { BaseMessage } from "@langchain/core/messages";
+import { StateGraph } from "../graph/state.js";
+import { Annotation } from "../graph/annotation.js";
+import { gatherIterator } from "../utils.js";
+import { StreamMode } from "../pregel/types.js";
+
+it("state graph", async () => {
+  const state = Annotation.Root({
+    foo: Annotation({
+      reducer: (state: string[], update: string | string[]) => {
+        return Array.isArray(update)
+          ? [...state, ...update]
+          : [...state, update];
+      },
+      default: () => [],
+    }),
+  });
+
+  const graph = new StateGraph(state)
+    .addSequence({
+      one: () => ({ foo: "one" }),
+      two: () => ({ foo: "two" }),
+      three: () => ({ foo: "three" }),
+    })
+    .addEdge("__start__", "one")
+    .compile();
+
+  const input = { foo: "bar" };
+
+  expectTypeOf(await gatherIterator(graph.stream(input))).toExtend<
+    { foo?: string[] | string | undefined }[]
```

Comment:
this one should be I think `{ ['one' | 'two' | 'three']: { foo?: string[] | string | undefined }}[]` ?